### PR TITLE
[release/0.6] applying latest changes from main

### PR DESCRIPTION
**Pull request by bfineran**
mostly updates and fixes for YOLO and BERT examples

### CODE REVIEW

File: examples/huggingface-transformers/squad_eval.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  -h, --help            show this help message and exit
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the eval on,
+                        defaults to all physical cores available on the system
+  -e {deepsparse,onnxruntime}, --engine {deepsparse,onnxruntime}
+                        Inference engine backend to run eval on. Choices are
+                        'deepsparse', 'onnxruntime'. Default is 'deepsparse'
+  -s NUM_SOCKETS, --num-sockets NUM_SOCKETS
+                        For DeepSparse only. The number of physical cores to
+                        run the eval on. Defaults to None where is uses all
```
Comment (rahul-tuli) [inline examples/huggingface-transformers/squad_eval.py:42]:
```suggestion
                        run the eval on. Defaults to None where it uses all
```

File: examples/ultralytics-yolo/README.md [hunk 2]
```diff
@@ -14,24 +14,43 @@ See the License for the specific language governing permissions and
...
 sparsified models from [SparseZoo](https://github.com/neuralmagic/sparsezoo) or you can
 create your own using the 
-[Sparseml-Ultralytics-yolov3 integration](https://github.com/neuralmagic/sparseml/blob/main/integrations/ultralytics-yolov3/README.md).
+[Sparseml-Ultralytics-yolov5 integration](https://github.com/neuralmagic/sparseml/blob/main/integrations/ultralytics-yolov5/README.md).
 
 ## Installation
 The dependencies for this example can be installed using `pip`:
 ```bash
 pip3 install -r requirements.txt
 ```
```
Comment (rahul-tuli) [inline examples/ultralytics-yolo/README.md:33]:
```suggestion
Note: upgrade pip using `python -m pip install -U pip` before installing requirements
```

File: examples/ultralytics-yolo/README.md [hunk 3]
```diff
@@ -14,24 +14,43 @@ See the License for the specific language governing permissions and
...
+| Model Name     |      Stub      | Description |
+|----------|-------------|-------------|
+| yolov3-pruned-aggressive | zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned-aggressive_97 | Sparse YOLOv3 model train with full FP32 precision that recovers 97% of its baseline mAP |
+| yolov3-pruned_quant-aggressive | zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned_quant-aggressive_94 | Sparse INT8 quantized YOLOv3 model that recovers 94% of its baseline mAP |
+| yolov3-base | zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/base-none | Dense full precision YOLOv3-SPP model |
+
+
 ## Annotation Example
-`annotate.py` is a script for using YOLOv3 sparsified (and not sparsified) YOLOv3 models
+`annotate.py` is a script for using YOLO sparsified (and not sparsified) models
```
Comment (rahul-tuli) [inline examples/ultralytics-yolo/README.md:53]:
```suggestion
`annotate.py` is a script for using YOLO sparsified (or non-sparsified) models
```

File: examples/ultralytics-yolo/annotate.py [hunk 4]
```diff
@@ -23,9 +23,10 @@
                    [-c NUM_CORES] [-s NUM_SOCKETS] [-q] [--fp16]
                    [--device DEVICE] [--save-dir SAVE_DIR] [--name NAME]
                    [--target-fps TARGET_FPS] [--no-save]
+                   [--model-config MODEL_CONFIG]
                    model_filepath
 
-Annotate images, videos, and streams with sparsified YOLOv3 models
+Annotate images, videos, and streams with sparsified YOLO models
```
Comment (rahul-tuli) [inline examples/ultralytics-yolo/annotate.py:29]:
```suggestion
Annotate images, videos, and streams with sparsified or non-sparsified YOLO models
```

**Review by markurtz — DISMISSED** [review]

**Review by rahul-tuli — COMMENTED** [review]
LGTM @bfineran, left a few nits here and there. Good work!

File: examples/ultralytics-yolo/README.md [hunk 5]
```diff
@@ -14,24 +14,43 @@ See the License for the specific language governing permissions and
...
+The [SparseZoo](https://sparsezoo.neuralmagic.com/) contains both 
+baseline and models sparsified to different degrees of inference performance vs. baseline loss recovery.
+
+Each model in the SparseZoo has a specific stub that identifies it; any YOLO model stub in the SparseZoo can be used to
+run the following examples.
+
+
+| Model Name     |      Stub      | Description |
+|----------|-------------|-------------|
+| yolov3-pruned-aggressive | zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned-aggressive_97 | Sparse YOLOv3 model train with full FP32 precision that recovers 97% of its baseline mAP |
```
Comment (kevinaer) [inline examples/ultralytics-yolo/README.md:47]:
```suggestion
| yolov3-pruned-aggressive | zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned-aggressive_97 | Sparse YOLOv3 model trained with full FP32 precision that recovers 97% of its baseline mAP |
```

File: examples/ultralytics-yolo/README.md [hunk 6]
```diff
@@ -45,13 +64,13 @@ python annotate.py \
     --no-save  # webcam only
 ```
 
-In addition to webcam `--source` can take a path to a `.jpg` file, directory or glog pat
+In addition to webcam `--source` can take a path to a `.jpg` file, directory or glob path
```
Comment (kevinaer) [inline examples/ultralytics-yolo/README.md:67]:
```suggestion
In addition to webcam, `--source` can take a path to a `.jpg` file, directory or glob path
```

**Review by kevinaer — COMMENTED** [review]
Left some small nits for doc. Code looks great!

**Comment by bfineran** [comment]
Thanks @rahul-tuli @kevinaer. 

Synced offline with @rahul-tuli who will open a PR to make these fixes on main, then will land this rebase once those are included.

**Review by rahul-tuli — APPROVED** [review]
LGTM Ben!

**Review by kevinaer — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**